### PR TITLE
luajit: bump new version

### LINKED ISSUE #6068: ARM64 M1 LuaJIT build and testing problems
An umbrella for all problems appeared during the Apple M1 enabling
- [ ] #6065 
- [ ] #6066

### LINKED ISSUE #6097: JIT misbehaviour on M1 Macs
The following script yields invalid results with JIT enabled. When JIT engine is disabled, everything works fine.

```lua
local d = require('digest')
local t = { }

for _ = 1, 100 do
  local h = d.sha1_hex('')
  t[h] = t[h] and t[h] + 1 or 1
end

for k, v in pairs(t) do
  print(k, v)
end
```

Actual behavior:

```bash
./src/tarantool -e 'jit.on()' bug.lua
3939393939393939393939393939393939393939        86
da39a3ee5e6b4b0d3255bfef95601890afd80709        2
da39a3ee5e6b4b0d3d3d3d3d3d3d3d3d3d3d3d3d        1
3d3d3d3d3d3d3d3d3d3d3d3d3d3d3d3d3d3d3d3d        11
./src/tarantool -e 'jit.off()' bug.lua 
da39a3ee5e6b4b0d3255bfef95601890afd80709        100
```

Expected behavior:

```bash
./src/tarantool -e 'jit.on()' bug.lua 
da39a3ee5e6b4b0d3255bfef95601890afd80709        100
./src/tarantool -e 'jit.off()' bug.lua 
da39a3ee5e6b4b0d3255bfef95601890afd80709        100
```

### LINKED ISSUE #12480: Backporting LuaJIT upstream commits part 15
This ticket is an umbrella for all commits we planning to back-port in the next Q.
This list includes commits excluded from #12134 and may be updated later with some new.

general + x86/x64:
* [ ] [7152e154](https://github.com/LuaJIT/LuaJIT/commit/7152e154) Fix string.format for limited precision FP conversions.
* [ ] [54dc2fa5](https://github.com/LuaJIT/LuaJIT/commit/54dc2fa5) FFI: Add pre-declared int128_t, uint128_t, __int128 types. (**in progress** @Buristan)

Handling non-standard OSX C calling conventions (**in progress** @Buristan):
* [ ] [cf903edb](https://github.com/LuaJIT/LuaJIT/commit/cf903edb) FFI: Unify stack setup for C calls in interpreter.
* [ ] [83954100](https://github.com/LuaJIT/LuaJIT/commit/83954100) FFI/ARM64/OSX: Handle non-standard OSX C calling conventions.

arm64:
* [ ] [c2629764](https://github.com/LuaJIT/LuaJIT/commit/c2629764) ARM64: Fix pass-by-value struct calling conventions. (**in progress** @Buristan)

### PR DESCRIPTION
**Pull request by Buristan**
* test: introduce tests for debugging extensions
* lldb: refactor extension
* dbg: sort initialization of commands
* lldb: support full-range 64-bit lightuserdata
* dbg: generalize extension
* ci: introduce workflow to test debugger extension
* FFI: Unify stack setup for C calls in interpreter.
* FFI/ARM64/OSX: Handle non-standard OSX C calling conventions.
* ARM64: Fix pass-by-value struct calling conventions.
* FFI: Various ABI and calling convention fixes.
* FFI/MacOS: Fix calling convention for enums.
* test: fix libfficcall compilation for old GCC
* x86/x64: Change ipairs_aux to match JIT backend behavior.
* dbg: fix lj-stack command for LLDB
* dbg: fix DUALNUM detection for LLDB
* dbg: update help for the lj-arch command
* dbg: introduce lj-gco command
* dbg: introduce lj-bc, lj-func and lj-proto dumpers
* Fix corner case of os.time().
* Make check in os.time() consistent.
* dbg: introduce lj-ir, lj-jslots, lj-trace dumpers
* dbg: introduce lj-ctype command, extend cdata dump
* test: add verbose mode for debug extension tests
* Prevent sanitizer warnings for lj_tab_new*() and table.new().
* FFI/MacOS: Fix calling convention for on-stack varargs.
* test: skip lj-1470 test for unconfigured timezones

Closes #12480
Closes #6097
Closes #6068
Part of #4808

NO_DOC=LuaJIT submodule bump
NO_TEST=LuaJIT submodule bump

### CODE REVIEW

File: changelogs/unreleased/gh-12480-luajit-fixes.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+Backported patches from the vanilla LuaJIT trunk (gh-12480). The following
+issues were fixed as part of this activity:
+
+* Fixed JIT incorrect behaviour for vararg FFI functions on the aarch64 OSX
+  platform (gh-6097).
+* Fixed various ABI and calling convention misbehaviours for x64/aarch64
+  architectures.
+* Fixed `ipairs_aux()` to match JIT backend behaviour on x86/x64.
+* Fixed `os.time()` returning `-1`.
+* Fixed UBSan warnings for `table.new()`.
```
Comment (AArdeev) [inline changelogs/unreleased/gh-12480-luajit-fixes.md:12]: Backported patches from the vanilla LuaJIT trunk (gh-12480). The following
issues were fixed as part of this activity:

* Fixed incorrect JIT behavior for vararg FFI functions on the macOS AArch64
  platform (gh-6097).
* Fixed various ABI and calling convention issues for x64/AArch64
  architectures.
* Fixed `ipairs_aux()` to match JIT backend behavior on x86/x64.
* Fixed `os.time()` returning `-1`.
* Fixed UBSan warnings for `table.new()`.

**Review by AArdeev — APPROVED** [review]

File: changelogs/unreleased/gh-12480-luajit-fixes.md (hunk 1 above)
Comment (Buristan) [reply changelogs/unreleased/gh-12480-luajit-fixes.md:12]: Aleksey,
Thanks for the review!
Fixed your comment and force-pushed the branch.

```diff
diff --git a/changelogs/unreleased/gh-12480-luajit-fixes.md b/changelogs/unreleased/gh-12480-luajit-fixes.md
index 31d778666..6dcfacde8 100644
--- a/changelogs/unreleased/gh-12480-luajit-fixes.md
+++ b/changelogs/unreleased/gh-12480-luajit-fixes.md
@@ -3,10 +3,10 @@
 Backported patches from the vanilla LuaJIT trunk (gh-12480). The following
 issues were fixed as part of this activity:
 
-* Fixed JIT incorrect behaviour for vararg FFI functions on the aarch64 OSX
+* Fixed incorrect JIT behavior for vararg FFI functions on the macOS AArch64
   platform (gh-6097).
-* Fixed various ABI and calling convention misbehaviours for x64/aarch64
+* Fixed various FFI ABI and calling convention issues for x64/AArch64
   architectures.
-* Fixed `ipairs_aux()` to match JIT backend behaviour on x86/x64.
+* Fixed `ipairs_aux()` to match JIT backend behavior on x86/x64.
 * Fixed `os.time()` returning `-1`.
 * Fixed UBSan warnings for `table.new()`.
```

**Review by ligurio — APPROVED** [review]
LGTM

**Review by tmr-g — APPROVED** [review]
LGTM

**Review by ligurio — APPROVED** [review]

**Review by tmr-g — APPROVED** [review]
LGTM
